### DISCUSSION TITLE
Folder structure?

### DISCUSSION THREAD

**Discussion by rbronco** [DirtyRacer1337/Jellyfin.Plugin.PhoenixAdult#118] [Q&A] [answered]
I’ve been trying to use site folders to organize my videos, but I think that’s messing up the scraping. It seems like that makes it think there are a hundred different versions of the same video. Should everything be in a single folder? How can I organize my folder structure outside Emby?

Could the folder be used as the studio or site? That’s how I have been organizing.

**Comment by DirtyRacer1337** [comment, marked as answer]
Emby doesn't support that feature, Jellyfin only

**Reply by rbronco** [reply]
Does that mean you only get the file name and not the path of each file from Emby to scrape? I'm about to take a look at the code, but I am not really a programmer, so it'll probably just be a frustrating learning experience!

**Reply by DirtyRacer1337** [reply]
Not the file name, just some kind of `Name`. it's can be folder name or file name, that's all data that Emby gave for search
```csharp
using MediaBrowser.Model.Entities;
using System;

namespace MediaBrowser.Controller.Providers
{
    public class ItemLookupInfo : IHasProviderIds
    {
        public ItemLookupInfo();

        public string Name { get; set; }
        public string MetadataLanguage { get; set; }
        public string MetadataCountryCode { get; set; }
        public ProviderIdDictionary ProviderIds { get; set; }
        public int? Year { get; set; }
        public int? IndexNumber { get; set; }
        public int? ParentIndexNumber { get; set; }
        public DateTimeOffset? PremiereDate { get; set; }
        public bool IsAutomated { get; set; }
        public bool EnableAdultMetadata { get; set; }
    }
}
```

**Reply by rbronco** [reply]
Well, it compiles, but I get a bunch of these errors. With almost zero .NET and c# knowledge, I assume it can't find these dlls even though "assembled" shows up in the output. Do you have any hints? HtmlAgilityPack versions match, but Newtonsoft.Json shows 13.0.1 in VS.

Error App: LoaderException: Could not load file or assembly 'HtmlAgilityPack, Version=1.11.40.0, Culture=neutral, PublicKeyToken=bd319b19eaf3b43a'. The system cannot find the file specified.

Error App: LoaderException: Could not load file or assembly 'Newtonsoft.Json, Version=13.0.0.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aeed'. The system cannot find the file specified.

1>Added assembly 'bin\Release.Emby\netcoreapp3.1\PhoenixAdult.dll'
1>Added assembly 'bin\Release.Emby\netcoreapp3.1\HtmlAgilityPack.dll'
1>Added assembly 'bin\Release.Emby\netcoreapp3.1\FlareSolverrSharp.dll'
1>Added assembly 'bin\Release.Emby\netcoreapp3.1\HttpClient.Caching.dll'
1>Added assembly 'bin\Release.Emby\netcoreapp3.1\MihaZupan.HttpToSocks5Proxy.dll'
1>Added assembly 'bin\Release.Emby\netcoreapp3.1\Newtonsoft.Json.dll'
1>Merging 6 assembies to 'bin\Release.Emby\netcoreapp3.1\PhoenixAdult.dll'
1>Merge succeeded in 1.6969023 s
========== Rebuild All: 1 succeeded, 0 failed, 0 skipped ==========


I also saw a toggle in the settings for "Use FilePath for matching." Is this only for Jellyfin, like you mentioned above or is this a different thing?

**Reply by rbronco21** [reply]
I saw your version updates and tried again. I still get all the LoaderExceptions when I put it in Emby. Can you give me some help? If I can get it loaded I’ll be able to play around in the code. I can’t figure out this config junk. I tried downloading the source as well as using VS Community 2022 to git it. I get the same results either way. I should be able to open the solution and build it, right?

**Reply by DirtyRacer1337** [reply]
Idk, everything works for me and appveyor

**Reply by DirtyRacer1337** [reply]
Check versions in [PhoenixAdult.csproj](https://github.com/DirtyRacer1337/Jellyfin.Plugin.PhoenixAdult/blob/854dfb79507f0288e24ac1a1782592137dbdbcb1/Jellyfin.Plugin.PhoenixAdult/PhoenixAdult.csproj#L36)

**Reply by rbronco** [reply]
I found the issue. I used a post-build event to copy the dll to the server. Turns out the post-build event runs before the dll is assembled, so those assemblies actually were missing.

**Reply by rbronco** [reply]
Target framework also needs to be changed to .NET Core 3.1 for Emby to load it.

Emby log on startup:
<pre>Command line: /system/EmbyServer.dll -programdata /config -ffdetect /bin/ffdetect -ffmpeg /bin/ffmpeg -ffprobe /bin/ffprobe -restartexitcode 3
Operating system: Linux version 5.10.28-Unraid (root@Develop) (gcc (GCC) 9.3.0, GNU ld version 2.33.1-slack15) #1 SMP Wed Apr 7 08:23:18 PDT 2021

Framework: .NET Core 3.1.21
OS/Process: x64/x64
Runtime: system/System.Private.CoreLib.dll
Processor count: 4
Data path: /config
Application path: /system
.
.
.</pre>

**Reply by DirtyRacer1337** [reply]
No, it's shouldn't, https://docs.microsoft.com/en-us/dotnet/standard/net-standard?tabs=net-standard-2-1

**Reply by rbronco** [reply]
I'm sorry, I thought I had it working. I blew up the old project I had working and now I can't get it running again. I get different errors for each version. Here are the logs if you're curious. I will keep trying.
[NET Core 3.1.txt](https://github.com/DirtyRacer1337/Jellyfin.Plugin.PhoenixAdult/files/8055373/NET.Core.3.1.txt)
[NET Standard 2.0.txt](https://github.com/DirtyRacer1337/Jellyfin.Plugin.PhoenixAdult/files/8055375/NET.Standard.2.0.txt)
[NET Standard 2.1.txt](https://github.com/DirtyRacer1337/Jellyfin.Plugin.PhoenixAdult/files/8055376/NET.Standard.2.1.txt)
.

**Reply by DirtyRacer1337** [reply]
Check plugin latest version, I fixed this error

**Reply by rbronco** [reply]
Thanks, that did allow it to work, but only using .NET Core 3.1. Does this error mean that .NET Standard 2.1 wants to use System.Text.Json instead of Newtonsoft.Json?
<pre>Error App: LoaderException: Could not load file or assembly 'System.Text.Json, Version=5.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51'. The system cannot find the file specified.</pre>
[NET Standard 2.1.txt](https://github.com/DirtyRacer1337/Jellyfin.Plugin.PhoenixAdult/files/8064628/NET.Standard.2.1.txt)
[NET Core 3.1.txt](https://github.com/DirtyRacer1337/Jellyfin.Plugin.PhoenixAdult/files/8064629/NET.Core.3.1.txt)

**Reply by DirtyRacer1337** [reply]
Try with this bb3112e4f82f9529f1adfbec445f428d893d08b2, I disabled sentry for emby, I'm tired fighting with it
